### PR TITLE
Fixes fly not disabling on island exit

### LINKED ISSUE #23: The player can fly outside the island boundaries.
#### Screenshots and videos
https://streamable.com/0pgb7

#### Server
 - OS: **CentOS 7**
 - Java version: **8**

#### Plugins

image

##### BentoBox and Addons
image

####Extra:
Players cannot fly on the nether and end islands. Each time "Flying mode is turned on." says.

### PR DESCRIPTION
**Pull request by wellnesscookie**
The bug was that there was no check for user to see if he is in protection range or not after the countdown for fly disable.

Fixes #23 (Which is closed atm)

### CODE REVIEW

**Comment by wellnesscookie** [comment]
I have no idea what just happened, since I only commited to 1 file, somehow 3 extra commits are popping out too, but it says only 1 file is changed (as should be)

File: src/main/java/world/bentobox/islandfly/listeners/FlyListener.java [hunk 1]
```diff
@@ -43,9 +43,9 @@ public void onExitIsland(final IslandEvent.IslandExitEvent event) {
 
         final User user = User.getInstance(event.getPlayerUUID());
 
-        final Optional<Island> i = addon.getIslands().getIslandAt(user.getLocation());
+        final Island i = addon.getIslands().getIslandAt(user.getLocation()).orElse(null);
 
-        if (!i.isPresent()) return;
+        if (i == null) return;
```
Comment (tastybento) [inline src/main/java/world/bentobox/islandfly/listeners/FlyListener.java]: This code and the previous isPresent are exactly the same outcome.

**Comment by tastybento** [comment]
I'm going to merge this and then defector and write tests around it all to check all the edge cases. Thanks for the submission.

File: src/main/java/world/bentobox/islandfly/listeners/FlyListener.java (hunk 1 above)
Comment (wellnesscookie) [reply src/main/java/world/bentobox/islandfly/listeners/FlyListener.java]: Yes, tho I wasn't able to call #isPresent anymore because the Island is not of Optional type anymore.
But I added #orElse which could give the null.
